### PR TITLE
Main better mot logging

### PR DESCRIPTION
**Pull request by Simontwice**
# Description
refactor logging in LoggingLayer, as well as in Mixture of Tokens

# Checklist
   - [ ] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [ ] Are all variables set with logical defaults that are backward compatible?
   - [ ] I attached link to neptune if it was necessary
   - [ ] Have you successfully executed the linter and tests?
   - [ ] Are all functions concise and don't require splitting?
   - [ ] Is there documentation for arguments and complex logic?
   - [ ] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [ ] Are all functions placed in appropriate files?
   - [ ] Is the PR name meaningful and descriptive?
   - [ ] Is PR description provided, or unnecessary?
  
Other questions:
   - [ ] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [ ] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [ ] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/moe_layers/continuous_moe.py [hunk 1]
```diff
@@ -67,7 +65,7 @@ def rearrange_for_grouping(self, x):
         :return: x transposed so that the dimension to group over is second last
         """
         if self.sparsity_dim == 0:
-            x = torch.permute(x, [1, 0, 2])
+            x = torch.permute(x, [1, 0, 2])  # x = x.transpose(0,1)
```
Comment (Simontwice) [inline research/conditional/moe_layers/continuous_moe.py:68]:
```suggestion
            x = torch.permute(x, [1, 0, 2])
```

File: research/conditional/utils/layer_manager.py [hunk 2]
```diff
@@ -132,6 +136,8 @@ def log(self, verbosity_level):
             return self.log_light()
         elif verbosity_level == 2:
             return self.log_heavy()
+        elif verbosity_level == -1:
```
Comment (crewtool) [inline research/conditional/utils/layer_manager.py:139]: Maybe 3?

File: research/conditional/utils/layer_manager.py [hunk 3]
```diff
@@ -60,9 +61,12 @@ def log(self, step):
         ):
             if step % freq == 0:
                 verbosity_levels.append(level)
+        # if we are actually logging, we also want to log the time
+        if len(verbosity_levels) > 0:
+            verbosity_levels.append(-1)
```
Comment (crewtool) [inline research/conditional/utils/layer_manager.py:66]: I don't like this approach. How about we can have different levels of verbosity and higher level (or lower) consist new logs and from higher (or lower) levels

File: research/conditional/moe_layers/continuous_moe.py [hunk 4]
```diff
@@ -172,47 +173,48 @@ def log_heavy(self):
...
-            sample = sample.sort(descending=True).values
-            sample = sample.tolist()
-            fig = px.bar(x=range(len(sample)), y=sample, title=f"sample {i}")
-            log[f"merge_weights/sample_{i}"] = fig
-
-        for i, sample in enumerate(sample_logits_distros):
-            sample = sample.sort(descending=True).values
-            sample = sample.tolist()
-            fig = px.bar(x=range(len(sample)), y=sample, title=f"sample {i}")
-            log[f"merge_logits/sample_{i}"] = fig
```
Comment (crewtool) [inline research/conditional/moe_layers/continuous_moe.py:194]: we losing this image, is it intentional?

**Review by crewtool — COMMENTED** [review]
Left few comments

File: research/conditional/moe_layers/continuous_moe.py (hunk 4 above)
Comment (Simontwice) [reply research/conditional/moe_layers/continuous_moe.py:194]: yes we dropping this boi

File: research/conditional/utils/layer_manager.py (hunk 2 above)
Comment (Simontwice) [reply research/conditional/utils/layer_manager.py:139]: the current design is that timing info is always recorded when we have nonzero level of verbosity. I agree the design can be improved, but 3 is definitely not the way to go here

**Review by crewtool — APPROVED** [review]
LGTM

**Pull request merged**